In [8]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [9]:
df=pd.read_csv(r"C:\Users\user\Desktop\AIML_BC\Breast-Cancer-Detection-using-Machine-Learning-master\cancer dataset.csv")

In [10]:
from sklearn.preprocessing import LabelEncoder
labelencoder_Y=LabelEncoder()
df.iloc[:,1]=labelencoder_Y.fit_transform(df.iloc[:,1].values)

In [11]:
X=df.iloc[:,2:31].values #features that help us determine if patient has cancer or not
Y=df.iloc[:,1].values #this is the dataset containing our target variable which indicates diagnosis

In [14]:
from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test= train_test_split(X,Y, test_size=0.25, random_state=0)

In [18]:
from sklearn.preprocessing import StandardScaler
sc=StandardScaler()
X_train=sc.fit_transform(X_train)
X_test=sc.fit_transform(X_test)

In [16]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import confusion_matrix

In [35]:
dt=DecisionTreeClassifier(random_state=0)
dt.fit(X_train,Y_train)
cm=confusion_matrix(Y_test,dt.predict(X_test))
TP=cm[0][0]
TN=cm[1][1]
FP=cm[0][1]
FN=cm[1][0]
print("Accuracy of the Decision Tree Classifier is: {}".format((TP+TN)/(TP+TN+FP+FN)))


Accuracy of the Decision Tree Classifier is: 0.9020979020979021


In [25]:
from sklearn.ensemble import RandomForestClassifier

In [36]:
rf=RandomForestClassifier(random_state=0)
rf.fit(X_train,Y_train)
cm2=confusion_matrix(Y_test,rf.predict(X_test))
TP=cm2[0][0]
TN=cm2[1][1]
FP=cm2[0][1]
FN=cm2[1][0]
print("Accuracy of the Random Forest Classifier is: {}".format((TP+TN)/(TP+TN+FP+FN)))


Accuracy of the Random Forest Classifier is: 0.958041958041958


C:\Users\user\Anaconda3\lib\site-packages\sklearn\ensemble\forest.py:246: FutureWarning: The default value of n_estimators will change from 10 in version 0.20 to 100 in 0.22.
  "10 in version 0.20 to 100 in 0.22.", FutureWarning)


In [34]:
'''from sklearn.linear_model import LogisticRegression
log=LogisticRegression (random_state=0)
log.fit(X_train,Y_train)
cm3=confusion_matrix(Y_test,log.predict(X_test))
TP=cm3[0][0]
TN=cm3[1][1]
FP=cm3[0][1]
FN=cm3[1][0]
print("Accuracy of the Logistic Regression Classifier is: {}".format((TP+TN)/(TP+TN+FP+FN)))'''


'from sklearn.linear_model import LogisticRegression\nlog=LogisticRegression (random_state=0)\nlog.fit(X_train,Y_train)\ncm3=confusion_matrix(Y_test,log.predict(X_test))\nTP=cm3[0][0]\nTN=cm3[1][1]\nFP=cm3[0][1]\nFN=cm3[1][0]\nprint("Accuracy of the Logistic Regression Classifier is: {}".format((TP+TN)/(TP+TN+FP+FN)))'

#conclusion
--------------------
Here we can see that the Randomforestclassifier performs better than the DecisionTreeClassifier (Measured using the accuracy of the models: Decision Tree Classifier=0.90, Random Forest Classifier=0.96). DecisionTree classifier builds a single tree and the predictions are based on the rules generated from the Single tree.
On the other hand RandomForest is an ensemble model where multiple decision trees are created and the results from all the trees are considered for prediction. Hence Random Forest Tree has a higher performance.